CONTRIBUTING.md

```diff
@@ -0,0 +1,177 @@
+[security@invidious.io](mailto:security@invidious.io).**
+
+
+### Enhancement
+
+If you feel that some existing feature can be improved, you should open an "enhancement" issue.
+
+In your issue, describe what the Invidious currently does, what you don't like about it, and propose
+ways to improve that feature. If you can provide screenshots or drawings to support your explanation
+that's even better!
+
+Please be aware that Invidious is heavily aimed towards simplicity and being usable without
+Javascript. It means that we have to deal with compromises all over the place, and some features
+might not be as good as JavaScript-rich alternatives (like Freetube), that allows much more
+flexibility.
+
+
+### Feature resquests
+
+If you think that Invidious lacks some feature or another, you should open a "feature request".
+
+Do note that Invidious is and will remain a Youtube-only front-end. Requests to add support for
+other services (e.g: Bandcamp, [Odyssee](https://github.com/iv-org/invidious/issues/3022)) will be
+systematically rejected.
+
+
+
+## Translations
+
+Invidious is translated in many languages using Weblate:
+https://hosted.weblate.org/projects/invidious/.
+
+We recommend creating an account or connecting with one of the other authentication providers that
+Weblate supports (Github, GitLab, Google, etc..) for a better experience.
+
+We also accept translation updates using Pull requests, but please be aware that it represents more
+work for us to merge those.
+
+
+
+## Hosting an instance
+
+Another way to contribute to invidious is to host a [public instance]
+(https://instances.invidious.io/).
+
+To do so, you need a server (either a VPS or dedicated) with
+[enough ressources](https://docs.invidious.io/installation/#hardware-requirements) to handle the
+load. You'll also need a domain name with a valid TLS certificate (e.g provided by Let's Encrypt).
+
+Then, if your instance follows the
+[rules listed here](https://docs.invidious.io/instances/#rules-to-have-your-instance-in-this-list),
+you can request your instance to be added to the list by
+[creating an issue on the documentation repository](https://github.com/iv-org/documentation/issues).
+
+Once you've filled the form with your instance's informations, your instance will be added to our
+uptime monitor. From there, a probatory period of 30 days will start, to make sure that you can
+keep your instance online and up to date. Finally, your instance will be added to the list. We'll
+ask you to join our Matrix room, so that we can inform you of important updates and exchange with
+other maintainers.
+
+
+
+## Testing
+
+Once reviewed, the new features or bug fixes must be tested before being merged. In general, this
+can be achieved by running the following commands (change the PR number as required):
+```bash
+git clone https://github.com/iv-org/invidious
+cd invidious
+wget "https://github.com/iv-org/invidious/pull/4439.diff"
+git apply 4439.diff
```

Comment:
Can be done in another way by passing the URL directly to git apply (not sure of the syntax though)

Reply:
The short URL (`github.com`) just 301 redirects to the long one, so it's the same, except the short one is easier to remember and type.

"Patch" includes each commit separately, with message info and other overhead, where as "Diff" is the entire code change without commit information. It's pretty much the same in the end, `git apply` doesn't really care.

Reply:
The thing is that at one point pull/XX.diff was outdated but patch-diff wasn't, it's maybe solved but patch-diff was always correctly updated... so it seems safer to recommend its use.

Reply:
Alternatively, 

```
git fetch upstream pull/4591/head:<branch name>
git checkout <branch name>
```

Reply:
Alternatively,
```
git fetch upstream pull/4591/head:branchname
git checkout branchname
```